upload/views/uploads.py

```diff
@@ -39,6 +39,131 @@
 log = logging.getLogger(__name__)
 
 
+def create_upload(serializer, repository, commit, report, is_shelter_request, token):
+    version = (
+        serializer.validated_data["version"]
+        if "version" in serializer.validated_data
+        else None
+    )
+    archive_service = ArchiveService(repository)
+    # Create upload record
+    instance: ReportSession = serializer.save(
+        report_id=report.id,
+        upload_extras={"format_version": "v1"},
+    )
+
+    # Inserts mirror upload record into measurements table. CLI hits this endpoint
+    insert_coverage_measurement(
+        owner_id=repository.author.ownerid,
+        repo_id=repository.repoid,
+        commit_id=commit.id,
+        upload_id=instance.id,
+        uploader_used=UploaderType.CLI.value,
+        private_repo=repository.private,
+        report_type=report.report_type,
+    )
+
+    # only Shelter requests are allowed to set their own `storage_path`
+    if instance.storage_path is None or not is_shelter_request:
+        path = MinioEndpoints.raw_with_upload_id.get_path(
+            version="v4",
+            date=timezone.now().strftime("%Y-%m-%d"),
+            repo_hash=archive_service.storage_hash,
+            commit_sha=commit.commitid,
+            reportid=report.external_id,
+            uploadid=instance.external_id,
+        )
+        instance.storage_path = path
+        instance.save()
+    trigger_upload_task(repository, commit.commitid, instance, report)
+    activate_repo(repository)
+    send_analytics_data(commit, instance, version, token)
+    return instance
+
+
+def trigger_upload_task(repository, commit_sha, upload, report):
+    log.info(
+        "Triggering upload task",
+        extra=dict(
+            repo=repository.name,
+            commit=commit_sha,
+            upload_id=upload.id,
+            report_code=report.code,
+        ),
+    )
+    redis = get_redis_connection()
+    task_arguments = {
+        "commit": commit_sha,
+        "upload_id": upload.id,
+        "version": "v4",
+        "report_code": report.code,
+        "reportid": str(report.external_id),
+    }
+    dispatch_upload_task(task_arguments, repository, redis)
+
+
+def activate_repo(repository):
+    # Only update the fields if needed
+    if (
+        repository.activated
+        and repository.active
+        and not repository.deleted
+        and repository.coverage_enabled
+    ):
+        return
+    repository.activated = True
+    repository.active = True
+    repository.deleted = False
+    repository.coverage_enabled = True
+    repository.save(
+        update_fields=[
+            "activated",
+            "active",
+            "deleted",
+            "coverage_enabled",
+            "updatestamp",
+        ]
+    )
+
+
+def send_analytics_data(commit: Commit, upload: ReportSession, version, token):
```

Comment:
calling this `analytics_token` will hopefully make it more obvious what its purpose is, and that it is only used for analytics.

Reply:
Done!